pull_into_place/structures.py

```diff
@@ -215,6 +221,13 @@ def read_and_calculate(workspace, pdb_paths):
                         position = xyz_to_array(line[30:54].split())
                         distance = euclidean(restraint.position, position)
                         restraint_distances.append(distance)
+        filter_path = os.path.join(workspace.root_dir,'filters.txt')
+        for f in filter_list:
+            with open(filter_path,'r+') as file:
+                line_found = any(f in line for line in file)
+                if not line_found:
+                    file.seek(0, os.SEEK_END)
+                    file.write(f)
```

Comment:
Two things:

First, you shouldn't use `os.path.join(workspace.root_dir, 'filters.txt')` like this.  The problem is that this forces you to hard-code `filters.txt` in a number of places throughout the code.  I can't foresee wanting to change this name, but [DRY is still better than WET](https://en.wikipedia.org/wiki/Don't_repeat_yourself).  To get around this, you should add a `filter_list` property to the `Workspace` class, like this:

    @property
    filter_list(self):
        return os.path.join(self.root_dir, 'filters.txt')

Second, you should use YAML to read and write the filters list instead of rolling your own file format.  This would save ~5 lines of file IO code, some of which is duplicated in a few places, every time you read and write the file.  It would also reduce the chance of bugs, since YAML is much better tested than your code.  And it'll easily support more complicated structure, if we ever decide we need that.